### ISSUE TITLE
Redo recently merged feature

### ISSUE THREAD

**Issue by vlazar** [LeaVerou/awesomplete#16818]
@LeaVerou I'm starting work on key/value (aka array of objects) feature related PRs and issues and badly need you opinion on this.

Recently merged PR https://github.com/LeaVerou/awesomplete/pull/16794/files added `oginalEvent` to `awesomplete-select` event. However, the are several issues with it:

``` diff
...
-   select: function (selected) {
+   select: function (selected, originalEvent) {
...
    $.fire(this.input, "awesomplete-select", {
-                               }
+                               },
+                               originalEvent: originalEvent
```
1. It changes the public API
2. It only passes `oginalEvent` on `mousedown` event, but not on `keydown` (bug)
3. It passes the whole event object to `awesomplete-select` event for the original use case

> to know exact element that was clicked on when selecting an item in the dropdown

I think that:
1. Should be reverted, so we pass data via instance property in order not to change public API
2. Should be fixed and the same data should be passed to `awesomplete-select` caused by keypress
3. Only pass `event.target` instead on `event`, unless you see use case to pass the whole `mousedown` or `keydown` event. [I've discussed with PR author, and `event.target` is enough for his use case](https://github.com/LeaVerou/awesomplete/pull/16794#issuecomment-171731830)

**Comment by LeaVerou** [comment]
Agree with 2 & 3. :+1: 
Changing the public API is ok, as long as it's backwards compatible (and this is). But of course, it should be done carefully.

**Comment by vlazar** [comment]
Are you absolutely sure about 1? The `originalEvent` argument seems to have no meaning in the context of public API:

`completer.select(selected, originalEvent)`

It looks like it was only added because it was easier to pass an argument, and maybe author didn't realized that it also changes public API and `originalEvent` in the context of public API is meaningless.

**Comment by LeaVerou** [comment]
I wasn't saying that passing `originalEvent` around was a good idea. Just that (generally speaking) API changes are not the devil's spawn, as long as they're backwards compatible.
In this case I agree that it was a clumsy solution.

**Comment by vlazar** [comment]
Addressed 2 & 3 in #16819 (please do not merge, see questions in it)
